Cammie mode crashes with NoSuchFieldException

```
08-02 10:30:55.994  5969  5969 E AndroidRuntime: Process: be.ugent.zeus.hydra.open, PID: 5969
08-02 10:30:55.994  5969  5969 E AndroidRuntime: java.lang.InternalError
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at d7.i.a(UnsafeAccess.java:3)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at d7.h.<clinit>(Unknown Source:4)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at x6.g$a.<init>(OperatorObserveOn.java:10)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at x6.g.b(OperatorObserveOn.java:3)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at x6.c.b(OnSubscribeLift.java:5)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at t6.b.a(Observable.java:8)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at t6.b.b(Unknown Source:5)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at be.ugent.zeus.hydra.wpi.cammie.CammieActivity.loadMjpeg(CammieActivity.java:3)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at be.ugent.zeus.hydra.wpi.cammie.CammieActivity.onStart(CammieActivity.java:2)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1467)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.Activity.performStart(Activity.java:8082)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3732)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:201)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:288)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7870)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
08-02 10:30:55.994  5969  5969 E AndroidRuntime: Caused by: java.lang.NoSuchFieldException: No field producerIndex in class Ld7/h; (declaration of 'd7.h' appears in base.apk)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at java.lang.Class.getDeclaredField(Native Method)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at d7.i.a(UnsafeAccess.java:1)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        ... 23 more
```

On the open variant, obviously.

The retraced stacktrace seems to indicate it is something in the mjpeg library, but perhaps it is our fault since we obfuscate it.

```
08-02 10:30:55.994  5969  5969 E AndroidRuntime: Process: be.ugent.zeus.hydra.open, PID: 5969
08-02 10:30:55.994  5969  5969 E AndroidRuntime: java.lang.InternalError
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.internal.util.unsafe.UnsafeAccess.addressOf(UnsafeAccess.java:115)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.internal.util.unsafe.SpscArrayQueueProducerFields.<clinit>(SpscArrayQueue.java:42)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.<init>(OperatorObserveOn.java:126)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.internal.operators.OperatorObserveOn.call(OperatorObserveOn.java:74)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.internal.operators.OperatorObserveOn.call(OperatorObserveOn.java:40)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:44)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.Observable.subscribe(Observable.java:10423)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.Observable.subscribe(Observable.java:10390)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.Observable.subscribe(Observable.java:10195)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at be.ugent.zeus.hydra.wpi.cammie.CammieActivity.loadMjpeg(CammieActivity.java:118)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at be.ugent.zeus.hydra.wpi.cammie.CammieActivity.onStart(CammieActivity.java:106)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1467)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.Activity.performStart(Activity.java:8082)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3732)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:201)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:288)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7870)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
08-02 10:30:55.994  5969  5969 E AndroidRuntime: Caused by: java.lang.NoSuchFieldException: No field producerIndex in class Ld7/h; (declaration of 'd7.h' appears in base.apk)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at java.lang.Class.getDeclaredField(Native Method)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        at rx.internal.util.unsafe.UnsafeAccess.addressOf(UnsafeAccess.java:112)
08-02 10:30:55.994  5969  5969 E AndroidRuntime:        ... 23 more
```

Perhaps similar: https://stackoverflow.com/questions/57146355/how-to-fix-nosuchfieldexception-no-field-producerindex-in-release-build-when

Maybe fixed upstream someday https://github.com/niqdev/ipcam-view/issues/59#issuecomment-391341041

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -185,6 +185,7 @@ dependencies {
     implementation "androidx.lifecycle:lifecycle-common-java8:2.5.1"
     implementation "androidx.viewpager2:viewpager2:1.0.0"
     implementation 'androidx.room:room-runtime:2.4.3'
+    implementation 'com.artemzin.rxjava:proguard-rules:1.3.3.0'
     annotationProcessor 'androidx.room:room-compiler:2.4.3'
     implementation 'com.squareup.okhttp3:okhttp:4.10.0'
     implementation 'com.squareup.moshi:moshi:1.13.0'

